### PR TITLE
fix eigensolver strategy

### PR DESCRIPTION
**Pull request by armingeiser**
Set matrices to zero before Build().

This is necessary if the eigensolver is called several times on the same model part.
This happens e.g. in optimization.

The location of this reset is chosen according to the newton_raphson_strategy.

### CODE REVIEW

**Review by msandre — APPROVED** [review]

**Review by philbucher — APPROVED** [review]
also ok from my side, but please wait for approval from @qaumann

**Comment by philbucher** [comment]
ping @qaumann

**Comment by philbucher** [comment]
I talked to @qaumann , he is fine with this

**Pull request merged**
